packages/web-api/src/services/preApproval.ts

```diff
@@ -0,0 +1,379 @@
+import { PrismaClient, UserRole, PreApprovedUser, Prisma } from '@reefguide/db';
+import { BadRequestException, NotFoundException } from '../exceptions';
+
+/**
+ * Input type for creating a new pre-approved user
+ */
+export interface CreatePreApprovedUserInput {
```

Comment:
why not generate these from zod definitions in packages/types/src/api.ts with `z.infer`?
`CreatePreApprovedUserInputSchema`